components/patient-data/rest/src/test/java/org/phenotips/data/rest/internal/DefaultPatientByExternalIdResourceImplTest.java

```diff
@@ -0,0 +1,345 @@
+import org.xwiki.users.UserManager;
+
+import javax.inject.Provider;
+import javax.ws.rs.WebApplicationException;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.UriBuilder;
+import javax.ws.rs.core.UriInfo;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.mockito.Mockito.*;
+
+public class DefaultPatientByExternalIdResourceImplTest {
+    @Rule
+    public final MockitoComponentMockingRule<PatientByExternalIdResource> mocker =
+            new MockitoComponentMockingRule<PatientByExternalIdResource>(DefaultPatientByExternalIdResourceImpl.class);
+
+    @Mock
+    private Patient patient;
+
+    @Mock
+    private UriInfo uriInfo;
+
+    @Mock
+    private UriBuilder uriBuilder;
+
+    private Logger logger;
+
+    private PatientRepository repository;
+
+    private DomainObjectFactory factory;
+
+    private QueryManager qm;
+
+    private AuthorizationManager access;
+
+    private UserManager users;
+
+    private URI uri;
+
+    private XWikiContext context;
+
+    private DefaultPatientByExternalIdResourceImpl component;
+    
+    private final String eid = "eid";
+
+    private final String id = "id";
+
+    @Before
+    public void setUp() throws ComponentLookupException, URISyntaxException
+    {
+        MockitoAnnotations.initMocks(this);
+        this.uri = new URI("uri");
+
+        Execution execution = mock(Execution.class);
+        ExecutionContext executionContext = mock(ExecutionContext.class);
+        ComponentManager componentManager = this.mocker.getInstance(ComponentManager.class, "context");
+        when(componentManager.getInstance(Execution.class)).thenReturn(execution);
+        doReturn(executionContext).when(execution).getContext();
+        doReturn(mock(XWikiContext.class)).when(executionContext).getProperty("xwikicontext");
+
+        this.repository = this.mocker.getInstance(PatientRepository.class);
+        this.factory = this.mocker.getInstance(DomainObjectFactory.class);
+        this.qm = this.mocker.getInstance(QueryManager.class);
+        this.access = this.mocker.getInstance(AuthorizationManager.class);
+        this.users = this.mocker.getInstance(UserManager.class);
+        this.component = (DefaultPatientByExternalIdResourceImpl) this.mocker.getComponentUnderTest();
+        this.logger = this.mocker.getMockedLogger();
+        Provider<XWikiContext> provider = this.mocker.getInstance(XWikiContext.TYPE_PROVIDER);
+        this.context = provider.get();
+        ReflectionUtils.setFieldValue(this.component, "uriInfo", this.uriInfo);
+
+        doReturn(this.uriBuilder).when(this.uriInfo).getBaseUriBuilder();
+        when(this.patient.getId()).thenReturn(this.id);
+        when(this.users.getCurrentUser()).thenReturn(null);
+        when(this.patient.getDocument()).thenReturn(null);
```

Comment:
Instead of returning `null` you can mock a User:

    @Mock
    private User currentUser

Then you can do `doReturn(this.currentUser).when(this.users).getCurrentUser()`. Similarly, you can create a `private DocumentReference patientDocument` and do `doReturn(this.patientDocument).when(this.patient).getDocument()`.